src/Stratis.Bitcoin.Features.MemoryPool/MempoolBehavior.cs

```diff
@@ -318,7 +318,7 @@ private async Task ProcessInvAsync(INetworkPeer peer, InvPayload invPayload)
                 send.Inventory.Add(new InventoryVector(peer.AddSupportedOptions(InventoryType.MSG_TX), inv.Hash));
             }
 
-            if (peer.IsConnected)
+            if (peer.IsConnected && (send.Inventory.Count > 0))
```

Comment:
peer.IsConnected - it should probably be is handshaked instead

Reply:
or nothing, actually. since it looks like we will always be connected when we execute this method. 
and in case we are in the middle of disconnection this won't save us from the race condition